### PR TITLE
Adding the Rest of the Authentication Features

### LINKED ISSUE #145: Create Signup Page

### LINKED ISSUE #148: Add Logic to Register a User

### PR DESCRIPTION
**Pull request by NathanGrenier**
### 🛠 Proposed Changes:

Added the remaining authentication features (described below) and any other improvements.

### 📝 Details:

- [x] Added Register Page
- [x] Added Register User Logic

### 🔗 Related Issue(s):

- Resolves #145 
- Resolves #148

### CODE REVIEW

**Comment by briantkatch** [comment]
Ready for review @CarciDev @NathanGrenier

Right now the password reset link is logged to the console if EXEC_ENV == development. Once we decide on what we use to send email, this can easily be updated.

Also, it would be nice to be able to toast on the page that is the result of a redirection - do we want a global toast store?

**Comment by NathanGrenier** [comment]
> Also, it would be nice to be able to toast on the page that is the result of a redirection - do we want a global toast store?

Ya, https://github.com/NathanGrenier/SOEN-341/blob/1eef63449a139603474128b1f8c37ac7e6a984cc/src/routes/auth/register/%2Bpage.svelte#L22-L31

**Comment by briantkatch** [comment]
@NathanGrenier yeah, but the message is still found in the page itself, which means it's not suitable for toasting from any point in the code.

I'm thinking moving it to +layout.svelte with the message entirely saved to the store.

**Comment by NathanGrenier** [comment]
> @NathanGrenier yeah, but the message is still found in the page itself, which means it's not suitable for toasting from any point in the code, particularly server-side code.
> 
> I'm thinking moving it to +layout.svelte with the message entirely saved to the store.

The message's content is just an object. It has to be triggered client side though.

File: src/routes/auth/register/+page.server.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      }),
+      { path: "/" },
+    );
+
+    // Prevent open redirect, only allow destinations starting with /
+    if ((url.searchParams.get("destination") || "").indexOf("/") == 0) {
+      return redirect(302, url.searchParams.get("destination") || "/");
+    } else {
+      return redirect(302, "/");
+    }
```
Comment (NathanGrenier) [inline src/routes/auth/register/+page.server.ts:136]: The redirect after registering a user breaks the code that displays a toast to confirm the creation of a new user.

Comment (briantkatch) [reply src/routes/auth/register/+page.server.ts:136]: Yeah, I realize that, which is why I made the comment about a toast store in layout that can be triggered from anywhere in the code.

Even without changing that, I feel the login-and-redirect behaviour makes more sense than toasting and taking the user to the blank register form again.

**Review by CarciDev — APPROVED** [review]
Will make a bug about toasting

**Pull request merged**